src/UpupUploader.tsx

```diff
@@ -60,6 +64,13 @@ export type UploadFilesRef = {
     dynamicUploadFiles: (files: File[]) => Promise<string[] | null>
 }
 
+// Add this helper function at the top level
+const createFileWithId = (file: File) => {
+    return Object.assign(file, {
+        id: `${file.name}-${file.size}-${file.lastModified}-${Math.random()}`,
+    })
+}
+
```

Comment:
Change math.random() with an actual UUID.
(please make sure to not use math.random anywhere)

Reply:
Thanks @MedAmine1212, done